### PR TITLE
Add cht API as global to contact-summary

### PR DESCRIPTION
**Pull request by ernestoteo**
#214

### CODE REVIEW

File: src/harness.js
Comment (kennsippell) [inline src/harness.js:681]: important to handle this path. this is the most important path

File: test/contact-summary.spec.js
Comment (kennsippell) [inline test/contact-summary.spec.js:43]: the `harness.getContactSummary` interface is unchanged. this is not the correct fix for this test

File: src/core-adapter.js
Comment (kennsippell) [inline src/core-adapter.js:76]: can you remove these unneeded empty lines?

File: src/harness.js
Comment (kennsippell) [inline src/harness.js:677]: Can your naming be consistent?  Let's call this thing the same thing everywhere.

```suggestion
    const chtScriptApi = this.coreAdapter.chtScriptApi(this.options.userRoles);
```

File: test/contact-summary.spec.js
Comment (kennsippell) [inline test/contact-summary.spec.js:41]: Can you use `to.deep.include` instead of `to.deep.eq` if you just want to validate a subset?

File: test/contact-summary.spec.js
Comment (kennsippell) [inline test/contact-summary.spec.js:96]: probably equally as useful to do something simpler like `expect(functionStub.args[0][3].v1).to.be.defined;`

this allows you to avoid importing assert and overasserting on the behaviour of the chtapi obj

File: docs/harness.js.html
Comment (kennsippell) [inline docs/harness.js.html:1]: nit: can you only commit the html files which have a change (other than the timestamp?)

File: test/contact-summary.spec.js
Comment (kennsippell) [inline test/contact-summary.spec.js:94]: can you add a test which actually uses the cht api end-to-end in the contact summary?

**Review by kennsippell — CHANGES_REQUESTED** [review]
great start

File: test/contact-summary.spec.js
Comment (kennsippell) [inline test/contact-summary.spec.js:121]: this is a unit test. an end to end test is from the perspective of a user. 

can you look at adding a test which has a contact-summary javascript file which uses the `cht.v1` for something?

Comment (ernestoteo) [reply test/contact-summary.spec.js:121]: @kennsippell, for sure, but this week I am mostly focused on OCA workflow. 
In order to do not block other tickets that are pending because of this we can merge this PR if it is OK, then I can add a test which has a contact-summary javascript file which uses the cht.v1 in another ticket.

File: test/cht-api.spec.js
Comment (kennsippell) [inline test/cht-api.spec.js:1]: can you add these into `contact-summary.spec.js` instead of making a new file?  it can be a new `describe('cht.v1'` section like this

File: test/cht-api.spec.js
Comment (kennsippell) [inline test/cht-api.spec.js:27]:
```suggestion
  it('#214 - cht API in contact summary - user has no permissions ', async () => {
```

File: src/core-adapter.js
Comment (kennsippell) [inline src/core-adapter.js:1]: Can you run `npm version patch` to bump version before merge?

**Review by kennsippell — CHANGES_REQUESTED** [review]
nice nice. almost there!

File: test/contact-summary.spec.js
Comment (kennsippell) [inline test/contact-summary.spec.js:124]: is it possible to just change [this](https://github.com/medic/cht-conf-test-harness/blob/341093bac194badd788387d3c6b0ad5dd548bfdf/test/contact-summary.spec.js#L9) to `project-with-source`? and avoid this boilerplate? 

This test class seems to mostly not care about what is in the contact-summary code

**Review by kennsippell — APPROVED** [review]

File: test/contact-summary.spec.js
Comment (ernestoteo) [reply test/contact-summary.spec.js:124]: Yes it is possible, then a change would be required [here](https://github.com/medic/cht-conf-test-harness/blob/341093bac194badd788387d3c6b0ad5dd548bfdf/test/contact-summary.spec.js#L90) because `project-with-source` does not have `contact-summary cards`  but `project-without-source` does have.

**Pull request merged**
